### PR TITLE
Now with 100% more Weather

### PR DESCRIPTION
**Pull request by falkyre**
Additions:  
* weather and alerts boards.
* weather observations data provided by Environment Canada (no api key) or OpenWeatherMap (api key)
* alerts data provided by EC (Canada Only) or National Weather Service (US only)
* See README.md under src/api/weather for more information

Updates:
* json validation against json schema (both in main script and also in nhl_setup)
* nhl_setup updated to do validation of config.json and create simple config with a single team from command line.  Also can create a simple config from within the app.  
* nhl_setup app updated to add new weather configuration entries 
* Updated nhl_setup README.md to document changes
* Updated json.load calls to catch errors and not print full trace.  Will exit app on bad json (both typo errors and validation errors)

### CODE REVIEW

**Comment by jeepdrvr71** [comment]
Has the weather been rolled into the general release?

**Comment by falkyre** [comment]
@jeepdrvr71 , I just submitted for @riffnshred to review today and if he's ok with everything, it should make general release.

File: src/main.py [hunk 1]
```diff
@@ -8,12 +8,16 @@
...
+from api.weather.owmWeather import owmWxWorker
+from api.weather.ecAlerts import ecWxAlerts
+from api.weather.nwsAlerts import nwsWxAlerts
 from renderer.matrix import Matrix
 import debug
 
 SCRIPT_NAME = "NHL-LED-SCOREBOARD"
 
-SCRIPT_VERSION = "1.1.3"
+SCRIPT_VERSION = "1.2.0 - Wx version"
```
Comment (falkyre) [inline src/main.py:20]: This was just my holder value for the version, you can decide what the version should be.

**Comment by riffnshred** [comment]
As discussed, the options for the weather stuff needs to be moved in the boards configs like the Covid 19 ones. 

Cheers

**Comment by falkyre** [comment]
OK.  I think this is ready to go.  The weather configs have been moved under the boards config.  I fixed the nhl_setup to handle the weather configs and also added in the defaults for the covid_19 boards if you don't enable one of the boards.  Also, the update_checker has been added.  This uses your installed git to do a check against whatever branch you are on when you run the script on start and then every day at 3AM (arbitrarily picked as there shouldn't be games at this time so the update check shouldn't be affecting anything).  Most users will be on master branch.  If there's an update, a green bar will be added to the top on displayed boards.  This can be "disabled" by the hiding the network issue indicator on the clock.  Also, if you are using SSH keys to connect to github, you may get a ```Host Key Verification Failed``` error.  If this happens, append the known_hosts in your  pi user's .ssh directory to the root user's .ssh/known_hosts file.  That will fix the error.  Normal users that don't have a github account will not have this issue.  Users that have a github account and are connecting via HTTPS with a username and password has not been tested.

**Comment by riffnshred** [comment]
For some reason I can't test your branch because I get permission denied when I try to sync it on my Pi, as disussed, I'll close this pull request and you'll open a new one pointing to my Dev branch and I'll test it from there.

**Pull request closed without merging**